### PR TITLE
OCPQE-25162 add y upgrade jobs for 4.18

### PR DESCRIPTION
**Pull request by dis016**
[OCPQE-25162](https://issues.redhat.com//browse/OCPQE-25162) add y upgrade jobs for 4.18 from 4.17
/cc @jianlinliu @shellyyang1989 @jiajliu @JianLi-RH  @evakhoni

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 1]
```diff
@@ -0,0 +1,975 @@
...
+    namespace: ocp
+    tag: openstack-installer
+  tests-private-postupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.18"
+  tests-private-preupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:29]: 4.17？

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 2]
```diff
@@ -0,0 +1,975 @@
+base_images:
+  ansible:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:3]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+    namespace: ocp
+    tag: ansible
+  cli:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:7]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 4]
```diff
@@ -0,0 +1,975 @@
...
+    namespace: ocp
+    tag: cli
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
+  hypershift-operator:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:15]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 5]
```diff
@@ -0,0 +1,975 @@
...
+    namespace: ocp
+    tag: hypershift-operator
+  openstack-installer:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:19]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 6]
```diff
@@ -0,0 +1,975 @@
...
+  tests-private-postupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.18"
+  tests-private-preupg:
+    name: tests-private
+    namespace: ci
+    tag: "4.18"
+  tools:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:31]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 7]
```diff
@@ -0,0 +1,975 @@
...
+    namespace: ocp
+    tag: tools
+  upi-installer:
+    name: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:35]: should be the initial version, here it is 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 8]
```diff
@@ -0,0 +1,975 @@
...
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  latest:
+    release:
+      architecture: amd64
+      channel: candidate
+      version: "4.18"
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:52]: 4.17

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml (hunk 1 above)
Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:29]: Thanks @JianLi-RH, i will fix it.

Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:29]: looks `make update` updated in multiple places wrongly.

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:29]: yes, and please also check other files.

Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml]: fixed

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__multi-stable-4.18-upgrade-from-stable-4.17.yaml [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  latest:
+    prerelease:
+      architecture: multi
+      product: ocp
+      version_bounds:
+        lower: 4.17.0-0
```
[the remark this answers is not in the document]
Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__multi-stable-4.18-upgrade-from-stable-4.17.yaml]: fixed in latest commit.

**Comment by dis016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-ovn-winc-f7

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+      version: "4.18"
+  latest:
+    release:
+      architecture: amd64
+      channel: candidate
+      version: "4.17"
+  target:
+    release:
+      architecture: amd64
+      channel: candidate
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-stable-4.18-upgrade-from-stable-4.17.yaml:62]: Can the target setting catch the ec build for now ?

**Comment by dis016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-stable-4.18-upgrade-from-stable-4.17-gcp-ipi-disc-priv-oidc-fips-f28

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 11]
```diff
@@ -0,0 +1,975 @@
...
+      UPGRADE_PRE_RUN_TAGS: '@disconnected'
+    test:
+    - chain: openshift-upgrade-qe-test-disconnected
+    workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private
+- as: aws-ipi-disc-priv-sts-basecap-none-fips-f28
+  cron: 5 19 29 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      ADDITIONAL_ENABLED_CAPABILITIES: MachineAPI CloudCredential
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:112]:
```suggestion
      ADDITIONAL_ENABLED_CAPABILITIES: MachineAPI CloudCredential
```
ADDITIONAL_ENABLED_CAPABILITIES: MachineAPI CloudCredential CloudControllerManager Ingress

more info refer to https://github.com/openshift/release/pull/56205

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-nightly-4.18-upgrade-from-stable-4.17.yaml [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+    workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private
+- as: aws-ipi-disc-priv-sts-basecap-none-f28
+  cron: 2 8 15 * *
+  steps:
+    cluster_profile: aws-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: release:arm64-target
+    env:
+      ADDITIONAL_ENABLED_CAPABILITIES: MachineAPI CloudCredential
```
Comment (jiajliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-nightly-4.18-upgrade-from-stable-4.17.yaml:100]: the same with https://github.com/openshift/release/pull/56124/files#r1742880393

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml (hunk 11 above)
Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml]: Thanks @jiajliu, Fixed in latest commit.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-nightly-4.18-upgrade-from-stable-4.17.yaml (hunk 12 above)
Comment (dis016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__arm64-nightly-4.18-upgrade-from-stable-4.17.yaml]: Thanks @jiajliu, Fixed in latest commit.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**